### PR TITLE
Minor fix from clang-tidy

### PR DESCRIPTION
**Pull request by justinchuby**
### Description

Minor fix from clang-tidy
1. Make `type_str` not const because `clang-tidy` says this prevents it from being moved.
2. Change `GetErrorWithNodeInfo` to take a const reference of `std::runtime_error`

### Motivation and Context

Minor adjustment to tidy code

### CODE REVIEW

**Review by jcwchen — APPROVED** [review]
Thank you for catching these. Just wondering are these the only ones that clang-tidy disallows in the whole codebase?

https://github.com/onnx/onnx/issues/4459 It would be great if we can have a CI running it to catch it early.

File: onnx/shape_inference/implementation.cc [hunk 1]
```diff
@@ -1007,7 +1007,7 @@ std::vector<const TypeProto*> GraphInferencerImpl::doInferencing(
   return graph_output_types;
 }
 
-std::string GetErrorWithNodeInfo(const NodeProto& n, std::runtime_error err) {
+std::string GetErrorWithNodeInfo(const NodeProto& n, const std::runtime_error& err) {
```
Comment (jcwchen) [inline onnx/shape_inference/implementation.cc:1010]: onnx/shape_inference/implementation.h needs to be updated as well

**Review by jcwchen — COMMENTED** [review]
Thanks for the update. I think we can merge this PR: https://github.com/onnx/onnx/pull/5041 first and see whether the introduced clang-tidy can catch the warnings in this PR?

**Review by jcwchen — APPROVED** [review]

**Comment by jcwchen** [comment]
https://github.com/onnx/onnx/pull/5041 The new added clang-tidy does not work normally... Investigating.

**Review by gramalingam — APPROVED** [review]

**Comment by jcwchen** [comment]
I cannot solve the clang-tidy CI issue shortly and thus I will merge this PR first. Thanks for waiting!

**Pull request merged**